### PR TITLE
Add missing @Nullable annotations on org.gradle.internal types

### PR DESCRIPTION
**Pull request by eskatos**
As a follow up to this change https://github.com/gradle/gradle/pull/5813/files#diff-3b7f5d54ed8d807240b29e2421a18a4c that broke Kotlin consumers that use `JSR305` in strict mode.

### CODE REVIEW

File: subprojects/base-services/src/main/java/org/gradle/internal/Pair.java [hunk 1]
```diff
@@ -18,33 +18,39 @@
...
 
+import javax.annotation.Nullable;
+
 public final class Pair<L, R> {
 
     public final L left;
     public final R right;
 
-    private Pair(L left, R right) {
+    private Pair(@Nullable L left, @Nullable R right) {
```
Comment (bamboo) [inline subprojects/base-services/src/main/java/org/gradle/internal/Pair.java]: ❓ Is this constructor ever passed `null` values?

Comment (eskatos) [reply subprojects/base-services/src/main/java/org/gradle/internal/Pair.java]: Yes from `ProjectPropertySettingBuildLoader` via `Pair.of(l,r)`.

File: subprojects/base-services/src/main/java/org/gradle/internal/Pair.java [hunk 2]
```diff
@@ -18,33 +18,39 @@
 
 import com.google.common.base.Function;
 
+import javax.annotation.Nullable;
+
 public final class Pair<L, R> {
 
     public final L left;
```
Comment (bamboo) [inline subprojects/base-services/src/main/java/org/gradle/internal/Pair.java]: :x: Make fields `@Nullable` too?

File: subprojects/base-services/src/main/java/org/gradle/internal/Pair.java (hunk 1 above)
Comment (bamboo) [reply subprojects/base-services/src/main/java/org/gradle/internal/Pair.java]: Ok.

File: subprojects/base-services/src/main/java/org/gradle/internal/Pair.java (hunk 2 above)
Comment (eskatos) [reply subprojects/base-services/src/main/java/org/gradle/internal/Pair.java]: Done

**Pull request merged**